app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -34,6 +38,19 @@ class TransactionDetailsViewModel
                 true == ownerCredentialsRepository.retrieveCredentials()?.let { credentials ->
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
+
+    suspend fun validateSafeTxHash(transaction: TransactionDetails): Boolean {
+        return kotlin.runCatching {
+            val safe = safeRepository.getActiveSafe()
+            val safeTxHash = (transaction.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails).safeTxHash
+            val calculatedSafeTxHash = calculateSafeTxHash(safe!!.address, transaction)?.toHexString()?.addHexPrefix()
```

Comment:
Maybe it is worth having a method that encapsulates all these as `calculateSafeTxHash().asString()` ?

Reply:
curently it is used only in one place (tests not counting). I would wait and if we use it more than in one place - then this method could be added